### DISCUSSION TITLE
Firebase Test Lab for Android with profile or release mode

### DISCUSSION THREAD

**Discussion by ostk0069** [leancodepl/patrol#1001] [Q&A]
Currently, flutter integration test with Firebase Test Lab in Andriod is not working on profile mode or release mode, only debug mode does. This is not due to the patrol, but is anyone know the way to work with this.

|platform | configuration | works |
|---|---|---|
|Android| debug | yes |
|Android| profile | **no(I need this)** |
|Android| release | **no(I need this)** |
|iOS| debug | no |
|iOS| profile | yes |
|iOS| release | no |

The problem is running intgration test in debug mode is too buggy that this test will be too flaky.

**Comment by bartekpacia** [comment]
Hi @ostk0069 👋🏻  Thanks for asking!

This is already tracked as #346. You comment there and add a "👍" :)

**Comment by ostk0069** [comment]
What I want is profile / release mode on Android flutter integration test in Firebase Test Lab , not only in flutter integration test.

**Reply by bartekpacia** [reply]
Instead of doing:

```
./gradlew :app:assembleDebug -Ptarget="$(pwd)/../integration_test/example_test.dart"
./gradlew :app:assembleDebugAndroidTest
```

you can do:

```
./gradlew :app:assembleRelease -Ptarget="$(pwd)/../integration_test/example_test.dart"
./gradlew :app:assembleReleaseAndroidTest
```

and then upload to Firebase Test Lab:

```
gcloud firebase test android run --type instrumentation \
    --app build/app/outputs/apk/release/app-release.apk \
    --test build/app/outputs/apk/androidTest/release/app-release-androidTest.apk \
    --timeout 1m \
    --results-bucket=<RESULTS_BUCKET> \
    --results-dir=<RESULTS_DIRECTORY>
```

The above steps are modified version of [the steps described in the docs](https://patrol.leancode.co/ci).

This approach has a problem described in #898. We're working on solving it.
